lib/camt_parser/general/transaction.rb

```diff
@@ -3,8 +3,8 @@ class Transaction
     def initialize(xml_data, debit, amount = nil, currency = nil)
       @xml_data = xml_data
       @debit    = debit
-      @amount   = amount   || @xml_data.xpath('AmtDtls/TxAmt/Amt/text()').text
-      @currency = currency || @xml_data.xpath('AmtDtls/TxAmt/Amt/@Ccy').text
+      @amount   = parse_amount || amount
```

Comment:
The passed in `amount` and `currency` parameters are the once of the parent entry node, if it is defined there. According to `CamtParser::Entry#parse_transactions`, `amount` and `currency` are supposed to be defined if there is exactly one transaction node in the entry.
According to the specification that I references earlier, the `<Amt />`-tag (and with it, implicitly, the `currency` as well) is mandatory in a transaction detail. That is why the `amount` and `currency` defined in the transaction detail take precedence over the passed in parameters. If defined, they always are more specific than what could be defined in the parent scope (i.e. the entry). If `amount` and `currency` are undefined in the transaction detail, the values from the parent scope are used instead. That's the reasoning. I hope that makes sense.